### PR TITLE
Change how to retrieve environ project id

### PR DESCRIPTION
**Pull request by jasonwzm**

### CODE REVIEW

File: scripts/asm-installer/install_asm [hunk 1]
```diff
@@ -2083,15 +2083,8 @@ is_membership_crd_installed() {
...
-  local IDENTITY_PROVIDER
-  local IDENTITY
-  IDENTITY_PROVIDER="$(kubectl get memberships.hub.gke.io membership -o=json | jq .spec.identity_provider)"
-  IDENTITY="$(echo "${IDENTITY_PROVIDER}" | sed 's/^\"https:\/\/gkehub.googleapis.com\/projects\/\(.*\)\/locations\/global\/memberships\/\(.*\)\"$/\1 \2/g')"
-  if [[ -n "${IDENTITY}" ]]; then
-    read -r ENVIRON_PROJECT_ID HUB_MEMBERSHIP_ID <<EOF
-${IDENTITY}
-EOF
-  fi
+  HUB_MEMBERSHIP_ID="$(kubectl get memberships.hub.gke.io membership -o=json | jq .spec.owner.id | sed 's/^\"\/\/gkehub.googleapis.com\/projects\/\(.*\)\/locations\/global\/memberships\/\(.*\)\"$/\2/g')"
```
Comment (zerobfd) [inline scripts/asm-installer/install_asm:2086]: I think this might also be "container.googleapis.com" not just "gkehub.googleapis.com"

**Review by zerobfd — CHANGES_REQUESTED** [review]

**Review by jtrbs — APPROVED** [review]

**Comment by jtrbs** [comment]
I did a test on the new cluster with  new membership format. 
```
kubectl get memberships.hub.gke.io membership -o=json
{
    "apiVersion": "hub.gke.io/v1",
    "kind": "Membership",
    "metadata": {
        "annotations": {
            "kubectl.kubernetes.io/last-applied-configuration": "{\"apiVersion\":\"hub.gke.io/v1\",\"kind\":\"Membership\",\"metadata\":{\"annotations\":{},\"creationTimestamp\":null,\"labels\":{\"hub.gke.io/system\":\"true\"},\"name\":\"membership\"},\"spec\":{\"identity_provider\":\"https://container.googleapis.com/v1/projects/test-istio-tairan/locations/us-central1-a/clusters/test04\",\"owner\":{\"id\":\"//gkehub.googleapis.com/projects/414487806869/locations/global/memberships/test04\"},\"workload_identity_pool\":\"test-istio-tairan.svc.id.goog\"}}\n"
        },
        "creationTimestamp": "2021-03-31T21:56:31Z",
        "generation": 1,
        "labels": {
            "hub.gke.io/system": "true"
        },
        "managedFields": [
            {
                "apiVersion": "hub.gke.io/v1",
                "fieldsType": "FieldsV1",
                "fieldsV1": {
                    "f:metadata": {
                        "f:annotations": {
                            ".": {},
                            "f:kubectl.kubernetes.io/last-applied-configuration": {}
                        },
                        "f:labels": {
                            ".": {},
                            "f:hub.gke.io/system": {}
                        }
                    },
                    "f:spec": {
                        ".": {},
                        "f:identity_provider": {},
                        "f:owner": {
                            ".": {},
                            "f:id": {}
                        },
                        "f:workload_identity_pool": {}
                    }
                },
                "manager": "kubectl",
                "operation": "Update",
                "time": "2021-03-31T21:56:31Z"
            }
        ],
        "name": "membership",
        "resourceVersion": "2094",
        "selfLink": "/apis/hub.gke.io/v1/memberships/membership",
        "uid": "1ce31c82-d284-4926-91cc-4b6688922c38"
    },
    "spec": {
        "identity_provider": "https://container.googleapis.com/v1/projects/test-istio-tairan/locations/us-central1-a/clusters/test04",
        "owner": {
            "id": "//gkehub.googleapis.com/projects/414487806869/locations/global/memberships/test04"
        },
        "workload_identity_pool": "test-istio-tairan.svc.id.goog"
    }
}
```
Looks like for new Identity provider format container.googleapis.com*, spec.owner.id is still the same gkehub.googleapis.com

File: scripts/asm-installer/install_asm (hunk 1 above)
Comment (jtrbs) [reply scripts/asm-installer/install_asm:2086]: Have an example in the comment below. Looks like currently it is always the same "gkehub.googleapis.com"

**Comment by zerobfd** [comment]
If that's how it's going to be from now on then I'm fine with it, but let's confirm and make sure that it's not going to change unexpectedly like the identity provider format

**Comment by jtrbs** [comment]
@zerobfd yeah Michael has agreed on the current usage for Hub related preview features. They will follow up with us if there would be change in the future.

File: scripts/asm-installer/asm_vm [hunk 2]
```diff
@@ -913,12 +913,10 @@ is_cluster_registered() {
...
 
 populate_environ_info() {
-  if [[ -n "${ENVIRON_PROJECT_ID}" ]]; then return; fi
+  if [[ -n "${ENVIRON_PROJECT_ID}" && -n "${HUB_MEMBERSHIP_ID}" ]]; then return; fi
   if ! is_membership_crd_installed; then return; fi
   configure_kubectl
-  local IDENTITY_PROVIDER
-  IDENTITY_PROVIDER="$(kubectl get memberships.hub.gke.io membership -o=json | jq .spec.identity_provider)"
-  ENVIRON_PROJECT_ID="$(echo "${IDENTITY_PROVIDER}" | sed 's/^\"https:\/\/gkehub.googleapis.com\/projects\/\(.*\)\/locations\/global\/memberships\/\(.*\)\"$/\1/g')"
+  ENVIRON_PROJECT_ID="$(kubectl get memberships.hub.gke.io membership -o=json | jq .spec.workload_identity_pool | sed 's/^\"\(.*\).\(svc\|hub\).id.goog\"$/\1/g')"
```
Comment (JimmyCYJ) [inline scripts/asm-installer/asm_vm:919]: Just to confirm, this command returns the hub IDNS, is environ project id equal to IDNS? I thought environ project id is a project id.

Comment (JimmyCYJ) [reply scripts/asm-installer/asm_vm:919]: I see, you are correct. This extract project id from IDNS.

File: scripts/asm-installer/install_asm (hunk 1 above)
Comment (jasonwzm) [reply scripts/asm-installer/install_asm:2086]: This is confirmed. gkehub resource uri is the owner.

**Review by zerobfd — APPROVED** [review]

**Pull request merged**